packages/block-editor/src/components/block-list/block.js

```diff
@@ -79,6 +79,7 @@ const useDebouncedAccessibleBlockLabel = ( blockType, attributes, index, moverDi
 function BlockListBlock( {
 	mode,
 	isFocusMode,
+	isToolbarHidden,
```

Comment:
Minor: any reason you chose `hidden`? I wonder whether it should be inverted `isToolbarVisible` because to be visible is probably the natural state of most things (both in code and in the real world) so it feels more natural to express it that way.

Reply:
Sounds good to me. I chose this convention probably because explicitly hiding the Toolbar was what I was after. Inverting it makes total sense.

Reply:
I actually have second thoughts about this. It's because by calling `hideToolbar()` we want to be sure it remains hidden, but do we want to do the exact opposite for `showToolbar()`? I mean `showing` the Toolbar in this case means just _unhiding_ it and letting its visibility depend on other factors, like e.g. if the block is focused (show) or not (hide). So, checking if `isToolbarVisible` would be kinda unreliable for the `true` value because we don't force it as we do when `false` is returned.

TL;DR

`isToolbarVisible`:
- `true`: Maybe it is, maybe it isn't. It depends on other factors 🤷‍♂️
- `false`: It definitely is hidden.

Vs.

`isToolbarHidden`:
- `true`: It definitely is hidden.
- `false`: It's not hidden, which doesn't mean it's visible - check other factors.

Having said that, I'm wondering if we shouldn't use just one action for that: `hideToolbar(true/false)`. Calling with true (default) would set the `isHidden` flag to `true` and the opposite for calling with `false`. What do you think?

Reply:
> isToolbarHidden

I take it back. Now you outline the other factors `isToolbarHidden` seems more logical because we can say "it is hidden" when `true` but we are not saying "it is definietely visible" when `false. Keep as is.

> Having said that, I'm wondering if we shouldn't use just one action for that: hideToolbar(true/false). Calling with true (default) would set the isHidden flag to true and the opposite for calling with false. What do you think?

Not in favour of this. `hideToolbar(false)` feels a little cryptic. What does it do? Show the toolbar? Hide the toolbar and trigger some other unknown behaviour? I'd avoid it personally.

Reply:
> hideToolbar(false) feels a little cryptic

True. It's just that calling `showToolbar()` on the other hand sounds misleading. I was also thinking about `unhideToolbar()` which IMO would mean exactly what it would do, which is unset the `hidden` flag. How does that sound?